### ISSUE TITLE
Rule "ibm-required-array-properties-in-response" throws errors for schemas with 'allOf' keyword

### ISSUE THREAD

**Issue by emlaver** [IBM/openapi-validator#690]
We are seeing several errors for this rule, and each error references a schema that already contains the field in the `required` list.  
I'll use the below schema `SchedulerJobsResult` as an example:
```
---
additionalProperties: false
allOf:
  - $ref: "../../index.yaml#/components/schemas/CommonViewResult"
  - additionalProperties: false
    properties:
      jobs:
        description: "Array of replication job objects."
        items:
          $ref: "../../index.yaml#/components/schemas/SchedulerJob"
        maxItems: 4294967296
        minItems: 0
        type: "array"
    type: "object"
required:
  - "jobs"
  - "offset"
  - "total_rows"
description: "Schema for a listing of replication scheduler jobs."

```
It throws the error:
```
  Message :   In a response body, an array field MUST NOT be optional
  Rule    :   ibm-required-array-properties-in-response
  Path    :   components.schemas.SchedulerJobsResult.allOf.1.properties.jobs
```

If I remove the `allOf` and just have one set of properties, I don't see the error.

This was tested with `ibm-openapi-validator` version `1.25.0`.

**Comment by padamstx** [comment]
Yep, I can see why the rule is complaining about "jobs".  Specifically, the rule is evaluating the second allOf element and finds that "jobs" is optional since - within the context of just that schema - no "required" field is found.

**Comment by padamstx** [comment]
The main "bug" is what I mentioned in my previous comment.  I've opened a PR that relaxes the rule slightly for an allOf element that does not include the "required" field.  This is simply a heuristic for whether or not we should bypass checks within an allOf element, and we'll see if it survives a code review :)

**Comment by padamstx** [comment]
YW...  @dpopp07 helped out also by suggesting some improvements to the rule's logic.

**Comment by dpopp07** [comment]
@emlaver no problem! Thanks for reporting the issue
